### PR TITLE
Improve byte formatting code

### PR DESCRIPTION
**Pull request by saagarjha**
- [ ] This change has been discussed with the author.
- [ ] It implements / fixes issue #.

---

I rewrote some of the byte formatting code, and thought I'd run it by a review before being merged. This lays the groundwork for some of the code that #2161 touches, and I wanted to centralize this logic. Note that I have tweaked the formatting slightly to comply with the SI and IEC standards.

### CODE REVIEW

File: iina/InspectorWindowController.swift [hunk 1]
```diff
@@ -139,7 +139,7 @@ class InspectorWindowController: NSWindowController, NSTableViewDelegate, NSTabl
         self.vsizeField.stringValue = "\(vwidth)\u{d7}\(vheight)"
 
         let fileSize = controller.getInt(MPVProperty.fileSize)
-        self.fileSizeField.stringValue = FileSize.format(fileSize, unit: .b)
+        self.fileSizeField.stringValue = "\(FloatingPointByteCountFormatter.string(fromByteCount: fileSize))B"
```
Comment (ErikMinekus) [inline iina/InspectorWindowController.swift]: File size should be formatted as binary.

File: iina/InspectorWindowController.swift [hunk 2]
```diff
@@ -171,10 +171,10 @@ class InspectorWindowController: NSWindowController, NSTableViewDelegate, NSTabl
       }
 
       let vbitrate = controller.getInt(MPVProperty.videoBitrate)
-      self.vbitrateField.stringValue = FileSize.format(vbitrate, unit: .b) + "bps"
+      self.vbitrateField.stringValue = FloatingPointByteCountFormatter.string(fromByteCount: vbitrate) + "bps"
 
       let abitrate = controller.getInt(MPVProperty.audioBitrate)
-      self.abitrateField.stringValue = FileSize.format(abitrate, unit: .b) + "bps"
+      self.abitrateField.stringValue = FloatingPointByteCountFormatter.string(fromByteCount: abitrate) + "bps"
```
[the remark this answers is not in the document]
Comment (saagarjha) [reply iina/InspectorWindowController.swift]: Yeah, I saw your pull request. I’m just changing this so the app is functional in the period between this being merged and your pull request being integrated.

[the remark this answers is not in the document]
Comment (ErikMinekus) [reply iina/InspectorWindowController.swift]: Ok, my bad.

File: iina/FloatingPointByteCountFormatter.swift [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+      }
+    }
+  }
+
+  static func string(fromByteCount byteCount: Int, prefixedBy prefixFactor: PrefixFactor = .none, digits: Int = 2, countStyle: CountStyle = .decimal) -> String {
+    let bytes = byteCount * prefixFactor.rawValue
+    let formatter = NumberFormatter()
+    formatter.maximumFractionDigits = digits
+    
+    if let prefixFactor = countStyle.prefixFactors.reversed().first(where: { bytes >= $0.rawValue }),
```
Comment (alejx) [inline iina/FloatingPointByteCountFormatter.swift]: `reversed().first { bytes >= $0.rawValue }` ?

Comment (saagarjha) [reply iina/FloatingPointByteCountFormatter.swift]: I had written it that way originally, but it doesn't look like this works?

<samp>
Trailing closure requires parentheses for disambiguation in this context
Replace ' { bytes >= $0.rawValue }' with '(where: { bytes >= $0.rawValue })'
</samp>

**Comment by saagarjha** [comment]
Merging. Feel free to revisit this if anyone comes up with a better name.

**Pull request merged**